### PR TITLE
Feature JSON

### PR DESCRIPTION
**Pull request by donpaul999**
- JSON tutorial complete
- list of messages is available on the first page

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -27,13 +27,38 @@ function addRandomGreeting() {
...
+    messagesListElement.innerHTML = '';
+    messagesListElement.appendChild(
+        createListElement(messages[0]));
+    messagesListElement.appendChild(
+        createListElement(messages[1]));
+    messagesListElement.appendChild(
+        createListElement(messages[2]));
+    messagesListElement.appendChild(
+        createListElement(messages[3]));
+    console.log(messages);
```
Comment (ramaroberto) [inline portfolio/src/main/webapp/script.js]: Don't leave console.log (s) after you're finishing debugging!

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -27,13 +27,38 @@ function addRandomGreeting() {
...
+    const messagesListElement = document.getElementById('messages-container');
+    messagesListElement.innerHTML = '';
+    messagesListElement.appendChild(
+        createListElement(messages[0]));
+    messagesListElement.appendChild(
+        createListElement(messages[1]));
+    messagesListElement.appendChild(
+        createListElement(messages[2]));
+    messagesListElement.appendChild(
+        createListElement(messages[3]));
```
Comment (ramaroberto) [inline portfolio/src/main/webapp/script.js]: You should use a for loop here, that way if you change the backend you don't need to change the frontend.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -27,13 +27,38 @@ function addRandomGreeting() {
...
 }
 
+/*
 async function getRandomNameUsingAsync() {
   const response = await fetch('/data');
   const name = await response.text();
  console.log('Adding name to dom: ' + name);
   document.getElementsByTagName('body').innerText = name; 
 }
+*/
```
Comment (ramaroberto) [inline portfolio/src/main/webapp/script.js]: Don't leave commented functions as the final part of code.

**Review by ramaroberto — CHANGES_REQUESTED** [review]
Feel free to ignore the comments if you're changing that same piece of code in a following PR; but please ACK them by telling me this.

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (donpaul999) [reply portfolio/src/main/webapp/script.js]: Solved

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (donpaul999) [reply portfolio/src/main/webapp/script.js]: Solved

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (donpaul999) [reply portfolio/src/main/webapp/script.js]: Solved

**Review by ramaroberto — APPROVED** [review]

**Pull request merged**